packages/app-project/src/components/AuthModal/components/RegisterForm/RegisterFormContainer.js

```diff
@@ -1,33 +1,126 @@
+import counterpart from 'counterpart'
 import { inject, observer } from 'mobx-react'
 import auth from 'panoptes-client/lib/auth'
+import { debounce } from 'lodash'
 import { func, shape } from 'prop-types'
 import React, { Component } from 'react'
+import en from './locales/en'
 
 import RegisterForm from './RegisterForm'
 
+const REMOTE_CHECK_DELAY = 1000
+counterpart.registerTranslations('en', en)
+
 class RegisterFormContainer extends Component {
   constructor() {
     super()
+    this.checkEmailForConflict = debounce(this.checkEmailForConflict, REMOTE_CHECK_DELAY)
+    this.checkNameForConflict = debounce(this.checkNameForConflict, REMOTE_CHECK_DELAY)
     this.onSubmit = this.onSubmit.bind(this)
+    this.validate = this.validate.bind(this)
+
+    this.state = {
+      error: ''
+    }
+  }
+
+  async validate (values) {
+    const { email, password, passwordConfirm, username } = values
+    let errors = {}
+
+    errors.username = await this.checkNameForConflict(username)
+    errors.email = await this.checkEmailForConflict(email)
```

Comment:
You're sending the register request twice here. Is there a reason not to register once, catch the error and then parse it?